packages/@ama-sdk/generator-sdk/src/generators/core/templates/swagger-codegen-typescript/src/main/resources/typescriptFetch/model/reviver.mustache

```diff
@@ -14,18 +14,17 @@ import { {{.}} } from '../{{#kebabCase}}{{.}}{{/kebabCase}}';
 import { utils, reviveArray, reviveMap, reviveDictionarizedArray } from '@ama-sdk/core';
 {{/isEnum}}{{/model}}{{/models}}
 {{/noEmptyLines}}{{#models}}{{#model}}{{^isEnum}}
-export function revive{{classname}}<T extends {{classname}} = {{classname}}>(data: undefined, dictionaries?: any): undefined;
-export function revive{{classname}}(data: {{classname}}, dictionaries?: any): {{classname}} {{#discriminator}}{{#vendorExtensions}}{{#x-discriminator-subtypes}}| {{.}}{{/x-discriminator-subtypes}}{{/vendorExtensions}}{{/discriminator}};
-export function revive{{classname}}(data: any, dictionaries?: any): {{classname}} {{#discriminator}}{{#vendorExtensions}}{{#x-discriminator-subtypes}}| {{.}}{{/x-discriminator-subtypes}}{{/vendorExtensions}}{{/discriminator}} | undefined;
-export function revive{{classname}}<T extends {{classname}}>(data: T, dictionaries?: any): T {{#discriminator}}{{#vendorExtensions}}{{#x-discriminator-subtypes}}| {{.}}{{/x-discriminator-subtypes}}{{/vendorExtensions}}{{/discriminator}};
-export function revive{{classname}}<T extends {{classname}}>(data: any, dictionaries?: any): T {{#discriminator}}{{#vendorExtensions}}{{#x-discriminator-subtypes}}| {{.}}{{/x-discriminator-subtypes}}{{/vendorExtensions}}{{/discriminator}} | undefined;
-export function revive{{classname}}<T extends {{classname}} = {{classname}}>(data: any, dictionaries?: any): T {{#discriminator}}{{#vendorExtensions}}{{#x-discriminator-subtypes}}| {{.}}{{/x-discriminator-subtypes}}{{/vendorExtensions}}{{/discriminator}} | undefined {
+export function revive{{classname}}<T extends {{classname}} = {{classname}}>(data: undefined, dictionaries?: any, logger = console): undefined;
```

Comment:
don't you need to explicit the `logger` type to `Logger`?
Otherwise it will take the type from `console` which is more complex